src/main/resources/redis.yml

```
@@ -17,4 +17,10 @@ spring:
     redis:
       host: ${REDIS_HOST}
       port: ${REDIS_PORT}
-      password: ${REDIS_PASSWORD}
\ No newline at end of file
+      password: ${REDIS_PASSWORD}
+
+---
+spring.config.activate.on-profile: prod
+
+---
+spring.config.activate.on-profile: test
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)